src/Makefile.test_util.include

```diff
@@ -10,22 +10,25 @@ EXTRA_LIBRARIES += \
 TEST_UTIL_H = \
     test/util/blockfilter.h \
     test/util/logging.h \
+    test/util/mining_utils.h \
```

Comment:
I think we do without the _utils suffix. No strong opinion, though.

Reply:
I used the suffix because we already have `wallet.h` twice. No strong opinion either though, so I am happy to change it after the CIs are through.

Reply:
Yeah, we use the include guards e.g. `BITCOIN_TEST_UTIL_WALLET_UTILS_H` to protect against duplicate includes and the `<>` import to protect against including the wrong path. So it should be fine to have three `wallet.h`

The travis CI passed and the appveyor ci is failing on master currently, so no need to wait for that.

Reply:
Ok, I renamed and rebased.
As to move-only, note that I removed one include from `util.cpp` (validationinterface) that I could not allocate to any of the helper functions.